huaweicloud/services/as/resource_huaweicloud_as_group.go

```diff
@@ -397,32 +398,60 @@ func refreshInstancesLifeStates(asClient *golangsdk.ServiceClient, groupID strin
 		if checkInService {
 			return allIns, "INSERVICE", err
 		}
-		logp.Printf("[DEBUG] Exit refreshInstancesLifeStates for %q!", groupID)
 		return allIns, "", err
 	}
 }
 
-func checkASGroupInstancesInService(ctx context.Context, asClient *golangsdk.ServiceClient, groupID string, insNum int, timeout time.Duration) error {
+func refreshGroupState(client *golangsdk.ServiceClient, groupID string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		asGroup, err := groups.Get(client, groupID).Extract()
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok {
+				return asGroup, "DELETED", nil
+			}
+			return nil, "ERROR", err
+		}
+		return asGroup, asGroup.Status, nil
+	}
+}
+
+func checkASGroupInstancesInService(ctx context.Context, client *golangsdk.ServiceClient, groupID string, insNum int, timeout time.Duration) error {
 	stateConf := &resource.StateChangeConf{
-		Pending: []string{"PENDING"},
-		Target:  []string{"INSERVICE"}, //if there is no lifecyclestatus, meaning no instances in asg
-		Refresh: refreshInstancesLifeStates(asClient, groupID, insNum, true),
-		Timeout: timeout,
-		Delay:   10 * time.Second,
+		Pending:      []string{"PENDING"},
+		Target:       []string{"INSERVICE"}, //if there is no lifecyclestatus, meaning no instances in asg
```

Comment:
this  comment looks useless and not in right place

Reply:
this comment will be deleted